File: flask/app/model/Processo.py
from extensions import db

class Processo(db.Model):
    processo_id = db.Column(db.Integer, primary_key=True)
    numero_processo = db.Column(db.String(50), nullable=False)
    valor_total = db.Column(db.Float, nullable=False)
    valor_entrada = db.Column(db.Float)

    # Foreign Key: links to Client.id
    cliente_id = db.Column(db.Integer, db.ForeignKey('cliente.cliente_id'), nullable=False)

File: flutter/app/lib/model/processo.dart
class Processo {
  final int processo_id;
  final double valor_total;
  final double valor_entrada;
  final String numero_processo;

  Processo({
    required this.processo_id,
    required this.valor_total,
    required this.valor_entrada,
    required this.numero_processo,
  });

  factory Processo.fromJson(Map<String, dynamic> json) {
    return Processo(
      processo_id: json['processo_id'] ?? 0,
      valor_total: json['valor_total'] ?? 0.0,
      valor_entrada: json['valor_entrada'] ?? 0.0,
      numero_processo: json['numero_processo'] ?? '',
    );
  }
}
